### PR TITLE
add tests (and fix bugs) for pools

### PR DESCRIPTION
**Pull request by dbeal-eth**
add tests to verify that the system imposed max debt limit is sufficiently
protecting the protocol from debt loss.

there were some bugs identified
during this process, but they have been fixed now.

### CODE REVIEW

File: packages/synthetix-main/contracts/storage/Vault.sol [hunk 1]
```diff
@@ -47,7 +47,7 @@ library Vault {
         /**
          * @dev The previous credit capacity of the vault (collateral - debt), when the system was last interacted with.
```
Comment (leomassazza) [inline packages/synthetix-main/contracts/storage/Vault.sol:48]: Comment outdated with old wording

File: packages/synthetix-main/contracts/storage/Pool.sol [hunk 2]
```diff
@@ -192,24 +198,31 @@ library Pool {
...
+        uint creditCapacityD18,
+        int debtD18
     ) internal view returns (int) {
+        Market.Data storage marketData = Market.load(marketId);
         uint minLiquidityRatioD18 = SystemPoolConfiguration.load().minLiquidityRatioD18;
 
         // Calculate the margin of debt that the market could incur to hit the system wide limit.
-        uint marginD18;
+        uint totalSharesD18 = self.vaultsDebtDistribution.totalSharesD18;
+        int valuePerShare = marketData.poolsDebtDistribution.getValuePerShare();
```
Comment (leomassazza) [inline packages/synthetix-main/contracts/storage/Pool.sol:210]: Add Decimal precision to variable name (valuePerShare)

**Review by leomassazza — CHANGES_REQUESTED** [review]
Check variable name nomenclature, and outdated comment. The rest looks great

File: packages/synthetix-main/contracts/storage/Pool.sol (hunk 2 above)
Comment (dbeal-eth) [reply packages/synthetix-main/contracts/storage/Pool.sol]: fixed

File: packages/synthetix-main/contracts/storage/Vault.sol (hunk 1 above)
Comment (dbeal-eth) [reply packages/synthetix-main/contracts/storage/Vault.sol]: thanks

**Review by leomassazza — APPROVED** [review]

**Pull request merged**